sdk/python/kfmd/metadata.py

```diff
@@ -171,11 +200,23 @@ def __init__(self,
     self.model_type = model_type
     self.training_framework = training_framework
     self.hyperparameters = hyperparameters
-    self.query = query
     self.labels = labels
     self._id = ""
     self._create_time = ""
 
+  def json(self):
```

Comment:
Some questions about this PR:
* How is this being kept in sync with the JSON schema definition?
* What happens if I want to log a TFX Artifact that represents a model?
* How can this `json` method be made to be scalable?

Reply:
Thanks for the review! I change `log_XXX` methods to one `log` method to log any artifact. It expects certain interface implemented. See the `log` method comment in the code.

> How is this being kept in sync with the JSON schema definition?

We can translate the JSON schema to a python class definition. But I don't find a good library to do that and defining it manually is simple now.

In terms of how to verify it is in sync. The JSON schema generates the backend artifact type and this method will get an error when saving the data. Again, an e2e test will be helpful but I can only test it locally now.

> What happens if I want to log a TFX Artifact that represents a model?

I made some change: As long as a python class implements two methods, it can be logged.

> How can this json method be made to be scalable?
First, `json()` is not required to send artifact. A simple `self._dict_` can be used instead of `json()`, but i don't want to expose underscored names such as `_id`, and `_create_time`.  

WDYT?

Reply:
I removed `json()` and use `__dict__` now, which looks more cleaner.